taxi/views.py

```diff
@@ -64,10 +64,28 @@ class CarListView(LoginRequiredMixin, generic.ListView):
 class CarDetailView(LoginRequiredMixin, generic.DetailView):
     model = Car
 
+    def post(self, *args, **kwargs):
+        action = self.request.POST.get("action")
+        car = Car.objects.get(id=kwargs["pk"])
+        if action == "add" and self.request.user not in car.drivers.all():
+            car.drivers.add(self.request.user)
+            car.save()
+            return HttpResponseRedirect(
+                reverse_lazy("taxi:car-detail", kwargs={"pk": car.pk})
+            )
+
+        if action == "delete" and self.request.user in car.drivers.all():
+            car.drivers.remove(self.request.user)
```

Comment:
1. use .exists() instead of "in" operator
2. no need to call .save() method after .add and .remove
3. you have a common part in both if statements, so don't duplicate the code